class.jetpack.php

```diff
@@ -3310,8 +3310,6 @@ public static function disconnect( $update_activated_state = true ) {
 		// Delete all the sync related data. Since it could be taking up space.
 		Sender::get_instance()->uninstall();
 
-		// Disable the Heartbeat cron
-		Jetpack_Heartbeat::init()->deactivate();
```

Comment:
That's a bit tricky. I get why we can't just deactivate all cron jobs as it may impact other plugins using this package, but in this case how can we make sure we don't leave active scheduled events laying around? Am I missing something?

Reply:
Here

https://github.com/Automattic/jetpack/pull/16285/files#diff-5d5dd765ee9393dec92c944543326154R175

We are only adding Jetpack info to the heartbeat if the connection is active. This is called the moment the data will be sent. So even if the heartbeat action triggers, no information will be sent.

In other words, the Heartbeat action will always trigger, but chances are it will have no data to send and will simply do nothing.

Does it answer your question?

Reply:
It does, thank you. 👍